src/test/java/org/terasology/logic/health/DamageEventTest.java

```diff
@@ -0,0 +1,148 @@
+import org.terasology.logic.health.event.*;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+import org.terasology.moduletestingenvironment.TestEventReceiver;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DamageEventTest extends ModuleTestingEnvironment {
+
+    private EntityManager entityManager;
+
+    private static final Logger logger = LoggerFactory.getLogger(DamageEventTest.class);
+
+    @Override
+    public Set<String> getDependencies() {
+        Set<String> modules = Sets.newHashSet();
+        modules.add("Health");
+        return modules;
+    }
+
+    @Before
+    public void initialize() {
+        entityManager = getHostContext().get(EntityManager.class);
+    }
+
+    @Test
+    public void damageTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 50;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoDamageEvent(10));
+        assertEquals(player.getComponent(HealthComponent.class).currentHealth, 40);
+
+        player.send(new DoDamageEvent(999));
+        assertEquals(player.getComponent(HealthComponent.class).currentHealth, 0);
+    }
+
+    @Test
+    public void damageEventSentTest() {
+        TestEventReceiver<BeforeDamagedEvent> receiver = new TestEventReceiver<>(getHostContext(), BeforeDamagedEvent.class);
+        List<BeforeDamagedEvent> list = receiver.getEvents();
+        assertTrue(list.isEmpty());
+
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 50;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoDamageEvent(10));
+        assertEquals(1, list.size());
+    }
+
+    @Test
+    public void damageModifyEventTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 50;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        TestEventReceiver<BeforeDamagedEvent> receiver = new TestEventReceiver<>(getHostContext(),
+                BeforeDamagedEvent.class,
+                (event, entity) -> {
+                    event.add(5);
+                    event.multiply(2);
+                });
+        // Expected damage value is ( initial:10 + 5 ) * 2 = 30
```

Comment:
Might want to keep an eye out for `=` vs `==`

Reply:
I think this line is just calculation, so `=` should work. 
Will keep an eye out for them

Reply:
It is just math, true - I tend to always use the code style setup when it is in code even if it would be legit otherwise ... 😁